juju/juju
# Feature/k8s controller jujud prepare

## Description of change

prepare for bootstrapping to k8s - machine agent part

## QA steps

- prepare microk8s cloud: <k8s-cloud>;
- bootstrap

```bash
$ juju bootstrap <k8s-cloud> --debug <controller-name>
```

- add a caas model;

```bash
$ juju add-model <model-name> <k8s-cloud>
```

- ensure storage

```bash
$ juju create-storage-pool mariadb-pv kubernetes storage-class=microk8s-hostpath
$ juju create-storage-pool operator-storage kubernetes storage-class=microk8s-hostpath
```
- deploy workload;

```bash
$ juju deploy cs:~juju/mariadb-k8s-0  --debug --storage database=500M,mariadb-pv
$ juju deploy cs:~juju/gitlab-k8s-0  --debug  --config juju-external-hostname=127.0.0.1.xip.io
$ juju relate mariadb-k8s gitlab-k8s
```

## Documentation changes

Not ready yet.

## Bug reference

Nil
